src/remote.c

```diff
@@ -271,7 +292,7 @@ static int create_internal(git_remote **out, git_repository *repo, const char *n
 	}
 
 	/* A remote without a name doesn't download tags */
-	if (!name)
+	if (is_anonymous)
```

Comment:
This is also different from what we did before, as now an unnamed remote with an associated repository will not fetch tags anymore. Is this intended?

Reply:
You're right, I went too far trying to make sense of the 3 types of remotes. Reverting.